### PR TITLE
Add audit-logs download command

### PR DESCRIPTION
**Pull request by yummybomb**
## Summary

Adds `kernel audit-logs download` for verified chunked audit-log exports.

- Requires an explicit, exclusive `[start, end)` range of at most 30 days.
- Downloads gzip-compressed JSONL with no alternate format modes.
- Walks the API cursor sequentially and verifies every source chunk checksum before processing it.
- Retries transient per-chunk failures (network errors, 429/5xx, checksum mismatches) up to 7 attempts (6 retries) with capped exponential backoff (~31s total) before failing the download.
- Uses the SDK's multi-value method exclusions to send the default GET exclusion and `--exclude-method` together to the API.
- Writes to an adjacent `.partial` file (replacing any stale one from a prior run) and publishes the final `.jsonl.gz` only after all chunks complete.
- Removes incomplete partial output after transfer failures and preserves a completed partial if final publication fails.
- Refuses to replace an existing output unless `--force` is passed; forced replacement happens only during the final rename.
- Chunk requests disable the SDK's built-in retries so the download's own retry policy is authoritative.

## Test plan

- [x] `go test ./...`
- [x] `go vet ./...`
- [x] Race-enabled audit-log tests
- [x] Windows amd64 test-binary cross-compile
- [x] Stacked method exclusions match search and download semantics
- [x] Finalization failures preserve completed partial and existing output files
- [x] Live download smoke test with stacked GET and POST exclusions
- [x] SDK retry suppression asserted against a real `httptest` server
- [x] Checksum-mismatch recovery and stale-partial overwrite tests

---

> [!NOTE]
> **Medium Risk**
> New CLI path writes sensitive audit data to disk with careful partial-file handling; behavior is well tested but touches export APIs and local file I/O.
> 
> **Overview**
> Adds **`kernel audit-logs download`** for bulk export of gzip-compressed JSONL over a required exclusive **`[start, end)`** window (max 30 days), reusing the same filter flags as search (including default GET exclusion via `auditLogExcludeMethods`).
> 
> The implementation pages through **`ExportChunk`**, validates each response with **`X-Content-Sha256`** and pagination headers, and applies its own retry policy (up to 7 attempts, capped backoff; SDK retries disabled) for transient failures. Output is written to a **`.partial`** file with restrictive permissions and only **renamed** to the final path after all chunks succeed; failed runs drop the partial, and a completed partial is kept if final rename fails. **`--force`** is required to overwrite an existing destination.
> 
> **`AuditLogsService`** gains **`ExportChunk`**; tests cover multi-chunk assembly, retries, checksum/header validation, overwrite semantics, and finalize edge cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 0cda7466dfff312d2ab3fddf28ccaad2a56022c7. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: cmd/audit_logs_download.go [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+		return params, fmt.Errorf("--start and --end are required")
+	}
+	start, _, err := parseAuditLogTime(in.Start)
+	if err != nil {
+		return params, fmt.Errorf("--start: %w", err)
+	}
+	end, _, err := parseAuditLogTime(in.End)
+	if err != nil {
+		return params, fmt.Errorf("--end: %w", err)
+	}
```
[the remark this answers is not in the document]
Comment (yummybomb) [reply cmd/audit_logs_download.go:195]: just made a pr to make the search exclusive on the end flag

File: cmd/audit_logs_download.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	want := res.Header.Get("X-Content-Sha256")
+	if want == "" {
+		return nil, nil, fmt.Errorf("response missing X-Content-Sha256 header")
+	}
+	sum := sha256.Sum256(body)
+	if got := hex.EncodeToString(sum[:]); got != want {
+		return nil, nil, fmt.Errorf("chunk checksum mismatch (got %s, want %s)", got, want)
+	}
+	return body, res.Header, nil
```
[the remark this answers is not in the document]
Comment (yummybomb) [reply cmd/audit_logs_download.go:161]: ExportChunk goes through the SDK’s requestconfig.Execute, which returns an error for every StatusCode >= 400. fetchAuditLogsChunk immediately returns on that SDK error, so a checksummed 4xx/5xx body cannot be appended.

File: cmd/audit_logs_download.go [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+		if !force {
+			return fmt.Errorf("%s already exists; pass --force to overwrite", outPath)
+		}
+	} else if !os.IsNotExist(err) {
+		return fmt.Errorf("inspect %s: %w", outPath, err)
+	}
+	if err := os.Rename(partialPath, outPath); err != nil {
+		return fmt.Errorf("finalize %s: %w", outPath, err)
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (yummybomb) [reply cmd/audit_logs_download.go:278]: Go’s Windows implementation of os.Rename calls MoveFileExW with MOVEFILE_REPLACE_EXISTING. Replacing an existing regular file is supported. It can still fail if the destination is open/locked or permissions deny replacement, but not merely because it already exists. The completed .partial is also preserved if finalization fails.

**Review by sjmiller609 — APPROVED** [review]
looks good, either as follow up or before merge, maybe you want to test interrupting one of the chunk downloads, e.g. cut your wifi then turn it back on, see if it can retry the missed chunks properly
